File: pset2/caesar/caesar.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<stdlib.h>
#include<ctype.h>

int main(int argc, string argv[]) {
    //argc when we run we do ./caesar 1 so ./{caesar} {1} argc = 2
    if(argc != 2) {
        printf("Usage: ./caesar key");
        // return error
        return 1;
    }

    int key = atoi(argv[1]);
    printf("Key: %d", key);

    string plaintext = get_string("plaintext:\t");
    int plainTextLength = strlen(plaintext);

    char cipherText[plainTextLength];
    int rotations = key % 26;
    int asciiIndex = 0;

    for(int i = 0; i < plainTextLength; i++) {
        if(isalpha(plaintext[i])) {
            asciiIndex = (int) plaintext[i] + rotations;

            if(isupper(plaintext[i]) && asciiIndex > 90) {
                asciiIndex = asciiIndex - 26;
            }

            if(islower(plaintext[i]) && asciiIndex > 122) {
                asciiIndex = asciiIndex - 26;
            }

            cipherText[i] = (char) asciiIndex;
        } else {
            cipherText[i] = plaintext[i];
        }
    }
    printf("ciphertext: %s\n", cipherText);
    return 0;
}

File: pset6/caesar/caesar.py
from sys import argv
import cs50


def main():
    # 1. get command line argument a hashed password
    # 2. password <= 5 digits
    #   2.1 password alphabetic chars (uppercase and/or lowercase)
    #   2.2 password examples MAjiD
    # 3. hash looks like {50}{GApilQSG3E2}
    #   3.1 50 is salt
    # 4. generate all permutations i.e lowercase, uppercase, mixed
    #   4.1 50GApilQSG3E2 == crypt(aaaaa,50)
    #   4.2 50GApilQSG3E2 == crypt(baaaa,50)
    #   4.3 50GApilQSG3E2 == crypt(zaaaa,50)
    #   4.4 50GApilQSG3E2 == crypt(abaaa,50)

    if len(argv) == 1 or len(argv) > 2:
        print("Usage: ./caesar key")
        exit("Usage: ./caesar key")

    key = int(argv[1])
    plainText = cs50.get_string("plaintext:\n")
    rotations = key % 26
    asciiIndex = 0
    cipherText = ""

    for c in plainText:

        if c.isalpha():
            asciiIndex = ord(c) + rotations

            if c.isupper() and asciiIndex > 90:
                asciiIndex = asciiIndex - 26

            if c.islower() and asciiIndex > 122:
               asciiIndex = asciiIndex - 26

            cipherText += chr(asciiIndex)
        else:
            cipherText += c

    print(f"ciphertext: {cipherText}\n");
    return 0

main()